### DISCUSSION TITLE
Building with namespace 33.1 completely breaks apk

### DISCUSSION THREAD

**Discussion by Kelsonk** [airsdk/Adobe-Runtime-Support#2097] [General]
I'm not even sure how to describe this, but changing the namespace of my application xml to 33.1 will result in a broken apk, and won't build the aab. It fails silently which is why it took me hours to figure out that it was the namespace.

Size of apk when running namespace 29.0 : ~35 mbs (correct)
Size of apk when running namespace 33.1: ~20 mbs (incorrect, immediately crashes when trying to run on phone)

I can post a screen of the file contents of the broken apk and a screen of what it should look like, I'm not sure what to do here.

Correct Apk
![correct](https://user-images.githubusercontent.com/6390370/186948193-049cf4c0-55e6-4391-9ee7-64a0b267b19c.png)

Busted Apk
![broken](https://user-images.githubusercontent.com/6390370/186947776-058a1547-18e9-40e9-a3d8-17399b482f46.png)

**Comment by ajwfrost** [comment]
My assumption here is that when you switch the namespace to 33.1, we are then using Gradle to build the APK file, and there's something about your configuration (most likely regarding an ANE and its resources, by the looks of things) that is causing a problem.

Short term fix: go back to using the "legacy" build method for APKs, by adding the flag `<BuildLegacyAPK>true</BuildLegacyAPK>` into the `<android>` section of your application descriptor file (valid only for 33.1 and above).

Longer term, or to actually create a bundle: you may need to update the ANE to incorporate resources within the appropriate folder/package name. As a starting point it's possible to update the configuration file (`AIRSDK/lib/adt.cfg`) to keep the Android Studio output, and then if you open this generated `AndroidStudioProject` folder up in Android Studio you may see what needs to be tweaked?

thanks

**Comment by Kelsonk** [comment]
Thanks a ton! Good enough for now, I'll worry about ane updates later, I'm a bit worried since a few of them are probably dead development wise.

**Comment by Kelsonk** [comment]
I deleted every ane that wasn't distriqt and update all my distriqt ane's and it's still a no-go, I think it'll be awhile before I start building aabs.
